File: KSConverter/ConfigFormats/PreFivePointFive.vb

'    Kernel Simulator  Copyright (C) 2018-2021  EoflaOE
'
'    This file is part of Kernel Simulator
'
'    Kernel Simulator is free software: you can redistribute it and/or modify
'    it under the terms of the GNU General Public License as published by
'    the Free Software Foundation, either version 3 of the License, or
'    (at your option) any later version.
'
'    Kernel Simulator is distributed in the hope that it will be useful,
'    but WITHOUT ANY WARRANTY; without even the implied warranty of
'    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
'    GNU General Public License for more details.
'
'    You should have received a copy of the GNU General Public License
'    along with this program.  If not, see <https://www.gnu.org/licenses/>.

Imports KS

Module PreFivePointFive

    'Pre-0.0.5.5 config format has "Kernel Version" as the first line.
    ''' <summary>
    ''' Takes configuration values and installs them to appropriate variables. Taken from config.vb version 0.0.5.2 with some removals that reflect this version
    ''' </summary>
    ''' <param name="PathToConfig">Path to pre-0.0.5.5 config (kernelConfig.ini)</param>
    Function ReadPreFivePointFiveConfig(PathToConfig As String) As Boolean
        Dim OldConfigReader As New IO.StreamReader(PathToConfig)
        Dim line As String = OldConfigReader.ReadLine
        Dim ValidFormat As Boolean
        Do While line <> ""
            If line.Contains("Kernel Version = ") Then
                ValidFormat = True
            End If
            If ValidFormat Then
                If line.Contains("Colored Shell = ") Then
                    If line.Replace("Colored Shell = ", "") = "False" Then
                        ColoredShell = False
                    End If
                End If
                If line.Contains("User Name Shell Color = ") Then
                    If ColoredShell = True Then UserNameShellColor = Val(CType([Enum].Parse(GetType(ConsoleColors), line.Replace("User Name Shell Color = ", "")), ConsoleColors))
                ElseIf line.Contains("Host Name Shell Color = ") Then
                    If ColoredShell = True Then HostNameShellColor = Val(CType([Enum].Parse(GetType(ConsoleColors), line.Replace("Host Name Shell Color = ", "")), ConsoleColors))
                ElseIf line.Contains("Continuable Kernel Error Color = ") Then
                    If ColoredShell = True Then ContKernelErrorColor = Val(CType([Enum].Parse(GetType(ConsoleColors), line.Replace("Continuable Kernel Error Color = ", "")), ConsoleColors))
                ElseIf line.Contains("Uncontinuable Kernel Error Color = ") Then
                    If ColoredShell = True Then UncontKernelErrorColor = Val(CType([Enum].Parse(GetType(ConsoleColors), line.Replace("Uncontinuable Kernel Error Color = ", "")), ConsoleColors))
                ElseIf line.Contains("Text Color = ") Then
                    If ColoredShell = True Then NeutralTextColor = Val(CType([Enum].Parse(GetType(ConsoleColors), line.Replace("Text Color = ", "")), ConsoleColors))
                ElseIf line.Contains("License Color = ") Then
                    If ColoredShell = True Then LicenseColor = Val(CType([Enum].Parse(GetType(ConsoleColors), line.Replace("License Color = ", "")), ConsoleColors))
                ElseIf line.Contains("Background Color = ") Then
                    If ColoredShell = True Then
                        BackgroundColor = Val(CType([Enum].Parse(GetType(ConsoleColors), line.Replace("Background Color = ", "")), ConsoleColors))
                    End If
                ElseIf line.Contains("Input Color = ") Then
                    If ColoredShell = True Then InputColor = Val(CType([Enum].Parse(GetType(ConsoleColors), line.Replace("Input Color = ", "")), ConsoleColors))
                ElseIf line.Contains("Listed command in Help Color = ") Then
                    If ColoredShell = True Then ListEntryColor = Val(CType([Enum].Parse(GetType(ConsoleColors), line.Replace("Listed command in Help Color = ", "")), ConsoleColors))
                ElseIf line.Contains("Definition of command in Help Color = ") Then
                    If ColoredShell = True Then ListValueColor = Val(CType([Enum].Parse(GetType(ConsoleColors), line.Replace("Definition of command in Help Color = ", "")), ConsoleColors))
                ElseIf line.Contains("Change Root Password = ") Then
                    If line.Replace("Change Root Password = ", "") = "True" Then
                        setRootPasswd = True
                    ElseIf line.Replace("Change Root Password = ", "") = "False" Then
                        setRootPasswd = False
                    End If
                ElseIf line.Contains("Set Root Password to = ") Then
                    If setRootPasswd = True Then
                        RootPasswd = line.Replace("Set Root Password to = ", "")
                    End If
                ElseIf line.Contains("Maintenance Mode = ") Then
                    If line.Replace("Maintenance Mode = ", "") = "True" Then
                        maintenance = True
                    ElseIf line.Replace("Maintenance Mode = ", "") = "False" Then
                        maintenance = False
                    End If
                ElseIf line.Contains("Prompt for Arguments on Boot = ") Then
                    If line.Replace("Prompt for Arguments on Boot = ", "") = "True" Then
                        argsOnBoot = True
                    ElseIf line.Replace("Prompt for Arguments on Boot = ", "") = "False" Then
                        argsOnBoot = False
                    End If
                ElseIf line.Contains("Clear Screen on Log-in = ") Then
                    If line.Replace("Clear Screen on Log-in = ", "") = "True" Then
                        clsOnLogin = True
                    ElseIf line.Replace("Clear Screen on Log-in = ", "") = "False" Then
                        clsOnLogin = False
                    End If
                ElseIf line.Contains("Show MOTD on Log-in = ") Then
                    If line.Replace("Show MOTD on Log-in = ", "") = "True" Then
                        showMOTD = True
                    ElseIf line.Replace("Show MOTD on Log-in = ", "") = "False" Then
                        showMOTD = False
                    End If
                ElseIf line.Contains("Simplified Help Command = ") Then
                    If line.Replace("Simplified Help Command = ", "") = "True" Then
                        simHelp = True
                    ElseIf line.Replace("Simplified Help Command = ", "") = "False" Then
                        simHelp = False
                    End If
                ElseIf line.Contains("Quiet Probe = ") Then
                    If line.Replace("Quiet Probe = ", "") = "True" Then
                        quietProbe = True
                    ElseIf line.Replace("Quiet Probe = ", "") = "False" Then
                        quietProbe = False
                    End If
                ElseIf line.Contains("Show Time/Date on Corner = ") Then
                    If line.Replace("Show Time/Date on Corner = ", "") = "True" Then
                        CornerTD = True
                    ElseIf line.Replace("Show Time/Date on Corner = ", "") = "False" Then
                        CornerTD = False
                    End If
                ElseIf line.Contains("MOTD = ") Then
                    MOTDMessage = line.Replace("MOTD = ", "")
                ElseIf line.Contains("Host Name = ") Then
                    HName = line.Replace("Host Name = ", "")
                ElseIf line.Contains("MOTD After Login = ") Then
                    MAL = line.Replace("MOTD After Login = ", "")
                End If
            End If
            line = OldConfigReader.ReadLine
        Loop
        OldConfigReader.Close()
        OldConfigReader.Dispose()
        Return ValidFormat
    End Function

End Module

File: KSConverter/ConfigFormats/PreFivePointFive.cs

// Kernel Simulator  Copyright (C) 2018-2022  Aptivi
// 
// This file is part of Kernel Simulator
// 
// Kernel Simulator is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
// 
// Kernel Simulator is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
// 
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.

#if !NETCOREAPP
using System;
using System.Diagnostics;
using System.IO;
using KS.ConsoleBase.Colors;
using KS.Kernel;
using KS.Misc.Writers.ConsoleWriters;
using KS.Shell;
using Terminaux.Colors;

namespace KSConverter
{
	static class PreFivePointFive
	{
		// Pre-0.0.5.5 config format has "Kernel Version" as the first line.
		/// <summary>
		/// Takes configuration values and installs them to appropriate variables. Taken from config.vb version 0.0.5.2 with some removals that reflect this version
		/// </summary>
		/// <param name="PathToConfig">Path to pre-0.0.5.5 config (kernelConfig.ini)</param>
		public static bool ReadPreFivePointFiveConfig(string PathToConfig)
		{
			try
			{
				var OldConfigReader = new StreamReader(PathToConfig);
				string line = OldConfigReader.ReadLine();
				var ValidFormat = default(bool);
				Debug.WriteLine("Reading pre-0.0.5.5 config...");
				while (!string.IsNullOrEmpty(line))
				{
					Debug.WriteLine($"Parsing line {line}...");
					if (line.Contains("Kernel Version = "))
					{
						Debug.WriteLine("Valid config!");
						ValidFormat = true;
					}
					if (ValidFormat)
					{
						if (line.Contains("Colored Shell = "))
						{
							if (line.Replace("Colored Shell = ", "") == "False")
							{
								Shell.ColoredShell = false;
							}
						}
						if (line.Contains("User Name Shell Color = "))
						{
							if (Shell.ColoredShell == true)
								KernelColorTools.UserNameShellColor = new Color(Convert.ToInt32(Enum.Parse(typeof(ConsoleColors), line.Replace("User Name Shell Color = ", ""))));
						}
						else if (line.Contains("Host Name Shell Color = "))
						{
							if (Shell.ColoredShell == true)
								KernelColorTools.HostNameShellColor = new Color(Convert.ToInt32(Enum.Parse(typeof(ConsoleColors), line.Replace("Host Name Shell Color = ", ""))));
						}
						else if (line.Contains("Continuable Kernel Error Color = "))
						{
							if (Shell.ColoredShell == true)
								KernelColorTools.ContKernelErrorColor = new Color(Convert.ToInt32(Enum.Parse(typeof(ConsoleColors), line.Replace("Continuable Kernel Error Color = ", ""))));
						}
						else if (line.Contains("Uncontinuable Kernel Error Color = "))
						{
							if (Shell.ColoredShell == true)
								KernelColorTools.UncontKernelErrorColor = new Color(Convert.ToInt32(Enum.Parse(typeof(ConsoleColors), line.Replace("Uncontinuable Kernel Error Color = ", ""))));
						}
						else if (line.Contains("Text Color = "))
						{
							if (Shell.ColoredShell == true)
								KernelColorTools.NeutralTextColor = new Color(Convert.ToInt32(Enum.Parse(typeof(ConsoleColors), line.Replace("Text Color = ", ""))));
						}
						else if (line.Contains("License Color = "))
						{
							if (Shell.ColoredShell == true)
								KernelColorTools.LicenseColor = new Color(Convert.ToInt32(Enum.Parse(typeof(ConsoleColors), line.Replace("License Color = ", ""))));
						}
						else if (line.Contains("Background Color = "))
						{
							if (Shell.ColoredShell == true)
							{
								KernelColorTools.BackgroundColor = new Color(Convert.ToInt32(Enum.Parse(typeof(ConsoleColors), line.Replace("Background Color = ", ""))));
							}
						}
						else if (line.Contains("Input Color = "))
						{
							if (Shell.ColoredShell == true)
								KernelColorTools.InputColor = new Color(Convert.ToInt32(Enum.Parse(typeof(ConsoleColors), line.Replace("Input Color = ", ""))));
						}
						else if (line.Contains("Listed command in Help Color = "))
						{
							if (Shell.ColoredShell == true)
								KernelColorTools.ListEntryColor = new Color(Convert.ToInt32(Enum.Parse(typeof(ConsoleColors), line.Replace("Listed command in Help Color = ", ""))));
						}
						else if (line.Contains("Definition of command in Help Color = "))
						{
							if (Shell.ColoredShell == true)
								KernelColorTools.ListValueColor = new Color(Convert.ToInt32(Enum.Parse(typeof(ConsoleColors), line.Replace("Definition of command in Help Color = ", ""))));
						}
						else if (line.Contains("Maintenance Mode = "))
						{
							if (line.Replace("Maintenance Mode = ", "") == "True")
							{
								Flags.Maintenance = true;
							}
							else if (line.Replace("Maintenance Mode = ", "") == "False")
							{
								Flags.Maintenance = false;
							}
						}
						else if (line.Contains("Prompt for Arguments on Boot = "))
						{
							if (line.Replace("Prompt for Arguments on Boot = ", "") == "True")
							{
								Flags.ArgsOnBoot = true;
							}
							else if (line.Replace("Prompt for Arguments on Boot = ", "") == "False")
							{
								Flags.ArgsOnBoot = false;
							}
						}
						else if (line.Contains("Clear Screen on Log-in = "))
						{
							if (line.Replace("Clear Screen on Log-in = ", "") == "True")
							{
								Flags.ClearOnLogin = true;
							}
							else if (line.Replace("Clear Screen on Log-in = ", "") == "False")
							{
								Flags.ClearOnLogin = false;
							}
						}
						else if (line.Contains("Show MOTD on Log-in = "))
						{
							if (line.Replace("Show MOTD on Log-in = ", "") == "True")
							{
								Flags.ShowMOTD = true;
							}
							else if (line.Replace("Show MOTD on Log-in = ", "") == "False")
							{
								Flags.ShowMOTD = false;
							}
						}
						else if (line.Contains("Simplified Help Command = "))
						{
							if (line.Replace("Simplified Help Command = ", "") == "True")
							{
								Flags.SimHelp = true;
							}
							else if (line.Replace("Simplified Help Command = ", "") == "False")
							{
								Flags.SimHelp = false;
							}
						}
						else if (line.Contains("Quiet Probe = "))
						{
							if (line.Replace("Quiet Probe = ", "") == "True")
							{
								Flags.QuietHardwareProbe = true;
							}
							else if (line.Replace("Quiet Probe = ", "") == "False")
							{
								Flags.QuietHardwareProbe = false;
							}
						}
						else if (line.Contains("Show Time/Date on Corner = "))
						{
							if (line.Replace("Show Time/Date on Corner = ", "") == "True")
							{
								Flags.CornerTimeDate = true;
							}
							else if (line.Replace("Show Time/Date on Corner = ", "") == "False")
							{
								Flags.CornerTimeDate = false;
							}
						}
						else if (line.Contains("MOTD = "))
						{
							Kernel.MOTDMessage = line.Replace("MOTD = ", "");
						}
						else if (line.Contains("Host Name = "))
						{
							Kernel.HostName = line.Replace("Host Name = ", "");
						}
						else if (line.Contains("MOTD After Login = "))
						{
							Kernel.MAL = line.Replace("MOTD After Login = ", "");
						}
					}
					line = OldConfigReader.ReadLine();
				}
				OldConfigReader.Close();
				OldConfigReader.Dispose();
				Debug.WriteLine($"Returning ValidFormat as {ValidFormat}...");
				return ValidFormat;
			}
			catch (Exception ex)
			{
				Debug.WriteLine($"Error while converting config! {ex.Message}");
				TextWriterColor.Write("  - Warning: Failed to completely convert config. Some of the configurations might not be fully migrated.", true, KernelColorTools.ColTypes.Warning);
				return false;
			}
		}

	}
}
#endif
